geopandas/geodataframe.py

```diff
@@ -76,7 +142,12 @@ def _set_geometry(self, col):
         if not isinstance(col, (list, np.ndarray, Series)):
             raise ValueError("Must use a list-like to set the geometry"
                              " property")
-        self.set_geometry(col, inplace=True)
+        if isinstance(col, GeoSeries):
+            self.set_geometry(col, inplace=True)
+        elif isinstance(col, (list, np.ndarray, Series)):
+            col = vectorized.from_shapely(col)
+            self.set_geometry(col.data, inplace=True)
```

Comment:
I've changed this, but I'm not sure that set_geometry will handle the VectorizedGeometry object.